Cython/Compiler/Builtin.py

```diff
@@ -810,9 +810,17 @@ def init_builtins():
 
     float_type = builtin_scope.lookup('float').type
     int_type = builtin_scope.lookup('int').type
-    bool_type  = builtin_scope.lookup('bool').type
+    #bool_type  = builtin_scope.lookup('bool').type
     complex_type  = builtin_scope.lookup('complex').type
 
+    # Most entries are initialized via "declare_builtin_type()"", except for "bool"
+    # which is apparently a special case because it conflicts with C++ bool.
+    # Here, we only declare it as builtin name, not as actual type.
+    bool_type = PyrexTypes.BuiltinObjectType(EncodedString('bool'), "((PyObject*)&PyBool_Type)", "PyLongObject")
+    bool_type.is_final_type = True
```

Comment:
Possibly not important but in `Symtab.BuiltinScope.declare_builtin_type` there's a:

```
if name == 'bool':
    type.is_final_type = True
```

I wonder if that's unneeded now?

Reply:
Yeah, that's where I got this from. It may have been unneeded for years, given that `bool` has been made special back when support for C++ `bool` was added. I'll leave it out of this PR and maybe remove it with a separate commit since it's less clear and dates back longer than the issues I'm solving here.